readthedocs/proxito/views/serve.py

```diff
@@ -225,7 +279,9 @@ def get(self, request, proxito_path, template_name='404.html'):
 
                 # TODO: decide if we need to check for infinite redirect here
                 # (from URL == to URL)
-                return HttpResponseRedirect(redirect_url)
+                response = HttpResponseRedirect(redirect_url)
+                cache.set('dirhtml_404+{proxito_path}', response, 60)
```

Comment:
Pretty sure we can't cache the object itself. We need to cache some string representation of it, then recreate the Python object.

Reply:
I tested this locally and it worked properly. I didn't find anything about HttpResponse not being serializable, but I'll take a deeper look for potential issues.

Reply:
Django's internals do the same: https://github.com/django/django/blob/4ed534758cb6a11df9f49baddecca5a6cdda9311/django/middleware/cache.py#L116

_Although_, that method check for some security concerns before caching the response, like `Cache-Control: private`, `Cookie` and caches only 200 and 304.

I don't think we need anyone of those in community at least. For commercial, I'm not sure yet, but I will think about.